source/material_model/reactive_fluid_transport.cc

```diff
@@ -263,8 +270,12 @@ namespace aspect
                 if (in.composition[q][porosity_idx] + porosity_change < 0)
                   porosity_change = -in.composition[q][porosity_idx];
 
-                if (c == bound_fluid_idx && this->get_timestep_number() > 0)
-                  reaction_rate_out->reaction_rates[q][c] = - porosity_change / fluid_reaction_time_scale;
+                if (fluid_solid_reaction_scheme != katz2003)
+                  {
+                    const unsigned int bound_fluid_idx = this->introspection().compositional_index_for_name("bound_fluid");
+                    if (c == bound_fluid_idx && this->get_timestep_number() > 0)
+                      reaction_rate_out->reaction_rates[q][c] = - porosity_change / fluid_reaction_time_scale;
+                  }
```

Comment:
I agree that this is not pretty, but I don't have a better suggestion, so let's leave it as is.